rust/src/traits/arithmetic/mod.rs

```diff
@@ -525,11 +525,123 @@ macro_rules! impl_float_inf_uni {
 }
 impl_float_inf_uni!(f64, f32; InfLn, inf_ln, neg_inf_ln, ln);
 impl_float_inf_uni!(f64, f32; InfLog2, inf_log2, neg_inf_log2, log2);
-impl_float_inf_uni!(f64, f32; InfExp, inf_exp, neg_inf_exp, exp);
 impl_float_inf_uni!(f64, f32; InfLn1P, inf_ln_1p, neg_inf_ln_1p, ln_1p);
 impl_float_inf_uni!(f64, f32; InfExpM1, inf_exp_m1, neg_inf_exp_m1, exp_m1);
 impl_float_inf_uni!(f64, f32; InfSqrt, inf_sqrt, neg_inf_sqrt, sqrt);
 
+// these implementations are expanded to catch errors in underflow.
+// when the input is very negative, resulting in an underflow, the output is min subnormal
+impl InfExp for f64 {
+    fn inf_exp(self) -> Fallible<Self> {
+        let not_finite = || {
+            err!(
+                Overflow,
+                concat!(
+                    "({}).",
+                    stringify!(inf_exp),
+                    "() is not finite. Consider tightening your parameters."
+                ),
```

Comment:
Here and below, I think `concat!` just takes literals? Do you want a `format!` in here?